### PR TITLE
test: datasets state isnotset test

### PR DESCRIPTION
**Pull request by jasonish**
A dataset state test to test for the first occurrence
of something being seen.

### CODE REVIEW

File: tests/datasets-state-isnotset/test.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+requires:
+  min-version: 5.0.0
+
+pcap: ../datasets-05-state/input.pcap
+
+args:
+  - --data-dir ./output
```
Comment (victorjulien) [inline tests/datasets-state-isnotset/test.yaml]: This is not portable
```
[15991] 5/10/2019 -- 08:24:13 - (suricata.c:1876) <Error> (ParseCommandLine) -- [ERRCODE: SC_ERR_LOGDIR_CMDLINE(117)] - The data directory "./output" supplied at the commandline (-d ./output) doesn't exist. Shutting down the engine.
```

**Comment by victorjulien** [comment]
Merged after some fixes for the output dir and missing requirements.

File: tests/datasets-state-isnotset/test.yaml (hunk 1 above)
Comment (jasonish) [reply tests/datasets-state-isnotset/test.yaml]: Ah, I see.  Maybe we should consider supporting environment vars in the test.yaml, should simplify this case.

**Comment by victorjulien** [comment]
On 05-10-19 22:02, Jason Ish wrote:
> *@jasonish* commented on this pull request.
> 
> ------------------------------------------------------------------------
> 
> In tests/datasets-state-isnotset/test.yaml
> <https://github.com/OISF/suricata-verify/pull/135#discussion_r331760090>:
> 
>> @@ -0,0 +1,13 @@
> +requires:
> +  min-version: 5.0.0
> +
> +pcap: ../datasets-05-state/input.pcap
> +
> +args:
> +  - --data-dir ./output
> 
> Ah, I see. Maybe we should consider supporting environment vars in the
> test.yaml, should simplify this case.

Yes this would be very useful. There are quite a few tests that could be
written without an explicit 'command' then.

**Pull request merged**